plugins/inputs/prometheus/kubernetes.go

```diff
@@ -96,6 +97,30 @@ func (p *Prometheus) startK8s(ctx context.Context) error {
 	return nil
 }
 
+func shouldScrapePod(pod *corev1.Pod, p *Prometheus) bool {
+	isCandidate := podReady(pod.Status.ContainerStatuses) &&
+		podHasMatchingNamespace(pod, p) &&
+		podHasMatchingLabelSelector(pod, p.podLabelSelector) &&
+		podHasMatchingFieldSelector(pod, p.podFieldSelector)
+
+	// Annotations always win over configuration. If annotation is not set
+	// scrape only if scrape_config is set
+	shouldScrape := func() bool {
+		switch p.MonitorKubernetesPodsMethod {
+		case MonitorMethodAnnotations: // must have 'true' annotation to be scraped
+			return pod.Annotations != nil && pod.Annotations["prometheus.io/scrape"] == "true"
+		case MonitorMethodSettings: // will be scraped regardless of annotation
+			return true
+		case MonitorMethodSettingsAndAnnotations: // will be scraped unless opts out with 'false' annotation
+			return pod.Annotations == nil || pod.Annotations["prometheus.io/scrape"] != "false"
+		default:
+			return true // should not get here
+		}
+	}()
```

Comment:
Why do we need this to be a function? Wouldn't this be more clear

```suggestion
	shouldScrape := true
	
	switch p.MonitorKubernetesPodsMethod {
	case MonitorMethodAnnotations: // must have 'true' annotation to be scraped
		shouldScrape =  pod.Annotations != nil && pod.Annotations["prometheus.io/scrape"] == "true"
	case MonitorMethodSettingsAndAnnotations: // will be scraped unless opts out with 'false' annotation
		shouldScrape = pod.Annotations == nil || pod.Annotations["prometheus.io/scrape"] != "false"
	}
```

Reply:
@srebhan , you are right, removed function.